File: src/api/src/routes/staff.ts
import { Router, Request, Response } from 'express';
import { db } from '../db/client';

const router = Router();

router.get('/', (_req: Request, res: Response) => {
  try {
    res.json(db.prepare('SELECT * FROM staff ORDER BY name').all());
  } catch (err) { res.status(500).json({ error: String(err) }); }
});

router.post('/', (req: Request, res: Response) => {
  try {
    const { facility_id, name, email, role } = req.body;
    const result = db.prepare(
      'INSERT INTO staff (facility_id, name, email, role) VALUES (?, ?, ?, ?)'
    ).run(facility_id, name, email, role);
    res.status(201).json({ id: Number(result.lastInsertRowid) });
  } catch (err) { res.status(500).json({ error: String(err) }); }
});

export default router;

File: src/api-python/routes/staff.py
from __future__ import annotations

from fastapi import APIRouter

from db.client import DbDep
from models import StaffCreate

router = APIRouter()


@router.get("/")
def list_staff(db: DbDep) -> list:
    return [dict(r) for r in db.execute("SELECT * FROM staff ORDER BY name").fetchall()]


@router.post("/", status_code=201)
def create_staff(body: StaffCreate, db: DbDep) -> dict:
    cur = db.execute(
        "INSERT INTO staff (facility_id, name, email, role) VALUES (?, ?, ?, ?)",
        (body.facility_id, body.name, body.email, body.role),
    )
    db.commit()
    return {"id": cur.lastrowid}
